### PR TITLE
refactor(web/engine): proper split-off of DOM-reliant code

### PR DESCRIPTION
**Pull request by jahorton**
Referring to our [Web-Core Refactor design doc](https://docs.google.com/document/d/1yXcxuVoXaT1nRFvGq9ws7U9GluZVI9t1VsGLEP0SbFE/edit?usp=sharing), it's almost time to complete "stage 1" for a headless web-core.  To facilitate the actual split, this PR splits a few classes into headless vs DOM-aware variants.

It's largely just moving code around and tying up the loose ends that process causes.

### CODE REVIEW

File: web/source/text/processor.ts [hunk 1]
```diff
@@ -161,11 +161,11 @@ namespace com.keyman.text {
           if(special == EmulationKeystrokes.Backspace) {
             // A browser's default backspace may fail to delete both parts of an SMP character.
             this.keyboardInterface.defaultBackspace(Lkc.Ltarg);
-          } else if(special) {
+          } else if(special || DefaultOutput.isCommand(Lkc)) { // Filters out 'commands' like TAB.
```
Comment (jahorton) [inline web/source/text/processor.ts:164]: This fixes a minor issue I noted in testing on master, where hardware TABs weren't changing the focused element.

File: web/source/dom/domDefaultOutput.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
+namespace com.keyman.text {
```
Comment (darcywong00) [inline web/source/dom/domDefaultOutput.ts:1]: Wouldn't the namespace be `com.keyman.dom`?

Comment (jahorton) [reply web/source/dom/domDefaultOutput.ts:1]: Kinda debated a bit on that... literally everything in the class is being done on `com.keyman.text`, so I went with the version that let me drop the namespace prefixing requirement.

I can change it back by adding a lot of `text.` prefixes everywhere in the file, if that's preferred.

File: web/source/utils/deepCopy.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+namespace com.keyman.utils {
+  /**
+   * Function     deepCopy
+   * Scope        Private
```
Comment (darcywong00) [inline web/source/utils/deepCopy.ts:4]: Is this still private?

File: web/source/dom/domDefaultOutput.ts (hunk 2 above)
Comment (darcywong00) [reply web/source/dom/domDefaultOutput.ts:1]: I got no preference

**Pull request merged**